### PR TITLE
Add a planning server package leveraging latest process managers

### PR DESCRIPTION
**Pull request by Levi-Armstrong**

### CODE REVIEW

File: tesseract_ros/tesseract_planning_server/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+  FILES_MATCHING PATTERN "*.h"
+  PATTERN ".svn" EXCLUDE
+ )
+
+install(DIRECTORY launch
+  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+)
+
+#if(CATKIN_ENABLE_TESTING)
```
Comment (marrts) [inline tesseract_ros/tesseract_planning_server/CMakeLists.txt:84]: Not sure if you want testing on here or not yet

File: tesseract_ros/tesseract_rosutils/src/conversions.cpp [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+}
+
+///**
+// * @brief Convert a cartesian pose to cartesian waypoint.
+// * @param pose The cartesian pose
+// * @param change_base A tranformation applied to the pose = change_base * pose
+// * @return WaypointPtr
+// */
+// inline tesseract_motion_planners::Waypoint::Ptr
+// toWaypoint(const geometry_msgs::Pose& pose,
```
Comment (marrts) [inline tesseract_ros/tesseract_rosutils/src/conversions.cpp:234]: How come all of these toWaypoint members are commented out?

**Review by marrts — COMMENTED** [review]
All my comments are just surface level changes/suggestions, everything seems to be working properly to me. Building and running on my system just fine.

File: tesseract_ros/tesseract_monitoring/include/tesseract_monitoring/tesseract_monitor_interface.h [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+   * @return Environment Shared Pointer, if nullptr it failed
+   */
+  template <typename S>
+  tesseract::Tesseract::Ptr getTesseract(const std::string& monitor_namespace) const
+  {
+    tesseract_msgs::GetEnvironmentInformation res;
+    res.request.flags = tesseract_msgs::GetEnvironmentInformationRequest::COMMAND_HISTORY |
+                        tesseract_msgs::GetEnvironmentInformationRequest::KINEMATICS_INFORMATION;
+
+    bool status = ros::service::call(R"(/)" + monitor_namespace + "/get_tesseract_information", res);
```
[the remark this answers is not in the document]
Comment (Levi-Armstrong) [reply tesseract_ros/tesseract_monitoring/include/tesseract_monitoring/tesseract_monitor_interface.h:133]: Done

File: tesseract_ros/tesseract_monitoring/include/tesseract_monitoring/tesseract_monitor_interface.h [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+   * @param monitor_namespace The namespace to extract the environment from.
+   * @return Environment Shared Pointer, if nullptr it failed
+   */
+  template <typename S>
+  tesseract_environment::Environment::Ptr getEnvironment(const std::string& monitor_namespace) const
+  {
+    tesseract_msgs::GetEnvironmentInformation res;
+    res.request.flags = tesseract_msgs::GetEnvironmentInformationRequest::COMMAND_HISTORY;
+
+    bool status = ros::service::call(R"(/)" + monitor_namespace + "/get_tesseract_information", res);
```
[the remark this answers is not in the document]
Comment (Levi-Armstrong) [reply tesseract_ros/tesseract_monitoring/include/tesseract_monitoring/tesseract_monitor_interface.h:97]: Done

File: tesseract_ros/tesseract_monitoring/launch/contact_monitor.launch [hunk 5]
```diff
@@ -14,18 +17,18 @@
...
   <arg name="contact_test_type" default="2"/>
   <arg name="contact_distance"/>
-  <arg name="publish_environment" default="false"/>
-  <arg name="publish_markers" default="false"/>
-
 
   <node pkg="tesseract_monitoring" type="tesseract_monitoring_contacts_node" name="tesseract_monitoring_contacts">
     <param name="robot_description" type="string" value="$(arg robot_description)"/>
-    <param name="plugin" type="string" value="$(arg plugin)"/>
+    <param name="descrete_plugin" type="string" value="$(arg descrete_plugin)"/>
```
[the remark this answers is not in the document]
Comment (Levi-Armstrong) [reply tesseract_ros/tesseract_monitoring/launch/contact_monitor.launch:23]: Fixed.

File: tesseract_ros/tesseract_monitoring/src/contact_monitor.cpp [hunk 6]
```diff
@@ -36,22 +36,30 @@ TESSERACT_COMMON_IGNORE_WARNINGS_POP
...
-ContactMonitor::ContactMonitor(const tesseract::Tesseract::Ptr& tess,
+const std::string ContactMonitor::DEFAULT_JOINT_STATES_TOPIC = "joint_states";
+const std::string ContactMonitor::DEFAULT_PUBLISH_ENVIRONMENT_TOPIC = R"(/tesseract_published_environment)";
+const std::string ContactMonitor::DEFAULT_MODIFY_ENVIRONMENT_SERVICE = R"(/modify_tesseract)";
+const std::string ContactMonitor::DEFAULT_PUBLISH_CONTACT_RESULTS_TOPIC = R"(/contact_results)";
+const std::string ContactMonitor::DEFAULT_PUBLISH_CONTACT_MARKER_TOPIC = R"(/contact_results_markers)";
+const std::string ContactMonitor::DEFAULT_COMPUTE_CONTACT_RESULTS_SERVICE = R"(/compute_contact_results)";
+
+ContactMonitor::ContactMonitor(std::string monitor_namespace,
+                               const tesseract::Tesseract::Ptr& tess,
```
[the remark this answers is not in the document]
Comment (Levi-Armstrong) [reply tesseract_ros/tesseract_monitoring/src/contact_monitor.cpp:49]: Done

File: tesseract_ros/tesseract_planning_server/CMakeLists.txt (hunk 1 above)
Comment (Levi-Armstrong) [reply tesseract_ros/tesseract_planning_server/CMakeLists.txt:84]: removed

File: tesseract_ros/tesseract_rosutils/src/conversions.cpp (hunk 2 above)
Comment (Levi-Armstrong) [reply tesseract_ros/tesseract_rosutils/src/conversions.cpp:234]: Ah, I need to clean those up. These are for the old waypoint type and no longer valid.

Comment (Levi-Armstrong) [reply tesseract_ros/tesseract_rosutils/src/conversions.cpp:234]: Removed.

File: tesseract_ros/tesseract_monitoring/src/environment_monitor.cpp [hunk 7]
```diff
@@ -344,17 +389,34 @@ void EnvironmentMonitor::environmentPublishingThread()
...
-  environment_subscriber_.shutdown();
-  ROS_INFO_NAMED(LOGNAME, "Stopped monitoring environment.");
+  monitored_environment_subscriber_.shutdown();
+  ROS_INFO_NAMED(monitor_namespace_, "Stopped monitoring environment.");
 }
 
-void EnvironmentMonitor::startMonitoringEnvironment(const std::string& monitor_environment_topic)
+const CurrentStateMonitorPtr& EnvironmentMonitor::getStateMonitor() const { return current_state_monitor_; }
+
+CurrentStateMonitorPtr& EnvironmentMonitor::getStateMonitorNonConst() { return current_state_monitor_; }
```
[the remark this answers is not in the document]
Comment (Levi-Armstrong) [reply tesseract_ros/tesseract_monitoring/src/environment_monitor.cpp:398]: Those were carried over from MoveIt. I will fix these.

**Pull request merged**